openml/config.py

```diff
@@ -400,10 +402,9 @@ def set_field_in_config_file(field: str, value: Any) -> None:
             # There doesn't seem to be a way to avoid writing defaults to file with configparser,
             # because it is impossible to distinguish from an explicitly set value that matches
             # the default value, to one that was set to its default because it was omitted.
-            value = config.get("FAKE_SECTION", f)  # type: ignore
-            if f == field:
-                value = globals()[f]
-            fh.write(f"{f} = {value}\n")
+            value = globals()[f] if f == field else config.get(f)  # type: ignore
+            if value is not None:
+                fh.write(f"{f} = {value}\n")
```

Comment:
Sorry for not being clear about the CLI fix. That's this bit of code.. it's used only by the CLI (I think). I understand that's not particularly clear since the other edit is in the `cli.py` file which you might reasonably expect  to also have the fix that was mentioned.

Basically, if `f!=field` then it would write the name of the variable to itself (e.g., `apikey = apikey`), instead of fetching the corresponding value from the configuration and re-writing that.